File: Midterm_example/src/main/java/pullUpConstuctorBody/After.java
package pullUpConstuctorBody;

/**
 *- Create constructor on superclass Employee
 *- Manager now use super() to call from superclass
 * instead of creating constructor
 */
class Employee{
    protected String name;
    protected String id;
    public Employee(String name, String id) {
        this.name = name;
        this.id = id;
    }
}

class Manager extends Employee{
    private int grade;
    /**
     * Calling super() on subclass
     */
    public Manager(String name, String id, int grade) {
        super(name, id);
        this.grade = grade;
    }
}

File: CSharpGenelization 3/CSharpGenelization/PullUpConstructorBody/After.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpGenelization.PullUpConstructorBody
{
    internal class After
    {
        //-Create constructor on superclass Employee
        //- Manager now only use base on constructor method 
        class Employee
        {
            protected String name;
            protected String id;
            public Employee(String name, String id)
            {
                this.name = name;
                this.id = id;
            }
        }

        class Manager : Employee
        {
            private int grade;
             //Doesn't have to call super() like in Java
             //just use base() on constructor method
            public Manager(String name, String id, int grade) : base(name, id)
            {
                this.grade = grade;
            }
    }
}
}
